src/cclib/parser/molproparser.py

```diff
@@ -916,6 +916,29 @@ def extract(self, inputfile, line):
                 self.atomcharges = {}
             self.atomcharges['mulliken'] = charges
 
+        if 'GRADIENT FOR STATE' in line:
+            reader = csv.reader(inputfile, delimiter=' ',
```

Comment:
The CSV reader can be avoided entirely. `line = next(inputfile)` or `next(inputfile)` is fine. For tokenization, see below.

Reply:
Yes you are right, that it can be avoided entirely in this case. Since most outputfiles are a mix of structured and unstructured data, I like to explicitly use a function that assumes structure, when parsing a table. Just for the human.